source/index.ts

```diff
@@ -121,6 +124,11 @@ class APIClient {
 
         throwIfNotOk(request, response);
 
+        if (waitSuccessRetries > 0) {
```

Comment:
How can this line be reached after `throwIfNotOk`? Does `uploadExisting` throw on `IN_PROGRESS` or not?

Reply:
Because the request doesn't fail, it returns a response, with `ok` as true, but the `uploadState` is `IN_PROGRESS`